internal/cli/cli.go

```diff
@@ -71,6 +73,19 @@ func Run(ctx context.Context, io *iostreams.IOStreams, args ...string) int {
 	cmd := root.New()
 	cmd.SetOut(io.Out)
 	cmd.SetErr(io.ErrOut)
+
+	// Special case for the launch command, support `flyctl launch args -- [subargs]`
+	// Where the arguments after `--` are passed to the scanner/dockerfile generator.
+	// This isn't supported natively by cobra, so we have to manually split the args
+	// See: https://github.com/spf13/cobra/issues/739
+	if len(args) > 0 && args[0] == "launch" {
```

Comment:
I think we should make it the default, it is useful for `fly m run` too and probably others.

Reply:
Can you say more?  `fly m run` doesn't run a scanner, so it isn't clear to me what subcommand would be the consumer of these arguments.

Reply:
Sorry Sam, I think what I want to do is a different issue after all.
 
`fly m run` accepts a command after the image name, think of it as the analogous of `docker run`

```
fly m run -a dan-sandbox alpine:latest sleep inf --aflag
Usage:
  flyctl machine run <image> [command] [flags]
...
Error: unknown flag: --aflag

...
 
but it works with `--` 🤯 

```
❯ fly m run -a dan-sandbox alpine:latest -- sleep inf --aflag
Searching for image 'alpine:latest' remotely...
image found: img_lj9x4d8n6repwo1k
Image: docker-hub-mirror.fly.io/library/alpine:latest
Image size: 3.4 MB

Success! A Machine has been successfully launched in app dan-sandbox
 Machine ID: 6e82740b42e287
 Instance ID: 01HTT5JDR0R7X7F21GWZP0C5AH
 State: created

 Attempting to start machine...

==> Monitoring health checks
No health checks found

Machine started, you can connect via the following private ip
  fdaa:0:5ea6:a7b:d33b:3fe3:7b83:2
```

and the flag was passed over:
```
❯ fly m exec 6e82740b42e287 'ps axf' |grep aflag
  305 root      0:00 sleep inf --aflag

```

Reply:
Hmmm...

```
% docker run --help

Usage:  docker run [OPTIONS] IMAGE [COMMAND] [ARG...]
```

vs:

```
% fly machine run --help
Run a machine

Usage:
  flyctl machine run <image> [command] [flags]
```

Note with Docker, options is before command, and with flyctl flags is after command, which would lead to the somewhat awkward:

```
  flyctl machine run <image> [command] [flags] -- [args]
```

That being said, what you tried already works (though not clearly documented):

```
  flyctl machine run <image> [flags] -- command [args]
```

Other than documentation, it isn't clear to me how to improve on that, while retaining backwards compatibility.